File: bitmovinanalyticscollector/src/main/java/com/bitmovin/bitmovinanalyticscollector/data/IEventDataDispatcher.java
package com.bitmovin.bitmovinanalyticscollector.data;

/**
 * Created by zachmanc on 12/19/17.
 */

public interface IEventDataDispatcher {
    public void enable();
    public void disable();
    public void add(EventData data);
    public void clear();
}

File: collector/src/main/java/com/bitmovin/analytics/data/IEventDataDispatcher.kt
package com.bitmovin.analytics.data

import com.bitmovin.analytics.dtos.AdEventData
import com.bitmovin.analytics.dtos.EventData

/**
 * Has to be greater or equal than Ingress side limit (currently 1000).
 */
const val SEQUENCE_NUMBER_LIMIT = 1000

interface IEventDataDispatcher {
    fun enable()

    fun disable()

    fun add(data: EventData)

    fun addAd(data: AdEventData)
}
